h2o-core/src/main/java/water/webserver/H2OHttpViewImpl.java

```diff
@@ -0,0 +1,194 @@
+package water.webserver;
+
+import org.apache.commons.io.IOUtils;
+import water.ExtensionManager;
+import water.api.DatasetServlet;
+import water.api.NpsBinServlet;
+import water.api.PostFileServlet;
+import water.api.PutKeyServlet;
+import water.api.RequestServer;
+import water.server.ServletUtils;
+import water.util.Log;
+import water.webserver.iface.H2OHttpConfig;
+import water.webserver.iface.H2OHttpView;
+import water.webserver.iface.RequestAuthExtension;
+
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+
+/**
+ * This is intended to be a singleton per H2O node.
+ */
+public class H2OHttpViewImpl implements H2OHttpView {
+  private static volatile boolean _acceptRequests = false;
+  private final H2OHttpConfig config;
+  private static final LinkedHashMap<String, Class<? extends HttpServlet>> SERVLETS = new LinkedHashMap<>();
+  static {
+    SERVLETS.put("/3/NodePersistentStorage.bin/*", NpsBinServlet.class);
+    SERVLETS.put("/3/PostFile.bin", PostFileServlet.class);
+    SERVLETS.put("/3/PostFile", PostFileServlet.class);
+    SERVLETS.put("/3/DownloadDataset", DatasetServlet.class);
+    SERVLETS.put("/3/DownloadDataset.bin", DatasetServlet.class);
+    SERVLETS.put("/3/PutKey.bin", PutKeyServlet.class);
+    SERVLETS.put("/3/PutKey", PutKeyServlet.class);
+    SERVLETS.put("/", RequestServer.class);
+  }
+
+  public H2OHttpViewImpl(H2OHttpConfig config) {
+    this.config = config;
+  }
+
+  public void acceptRequests() {
```

Comment:
There is a comment in `CustomHttpFilterTest` on line 33 that is obsolete because of this method.

Reply:
Doesn't need to be synchronized, once `_acceptRequests` is flipped to true, it never changes back.

Why the call to `notify`? Is this something that this PR needs?